components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/GeoServiceImpl.java

```diff
@@ -73,6 +75,15 @@
     public Response getGeoDeviceStats(@PathParam("deviceId") String deviceId,
                                       @PathParam("deviceType") String deviceType,
                                       @QueryParam("from") long from, @QueryParam("to") long to) {
+        //First, check whether the Geo Location service has been enabled in the cdmf-config.xml file
+        DeviceManagementConfig deviceManagementConfig = DeviceConfigurationManager.getInstance()
+                .getDeviceManagementConfig();
+        if (deviceManagementConfig != null) {
+            if(!deviceManagementConfig.getGeoLocationConfiguration().getPublishLocationOperationResponse()){
+                log.warn("Failed to fetch geo location stats since the GeoLocation service has not been enabled.");
+                return Response.status(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()).build();
```

Comment:
Shall we remove this log.warn() message since this log message will print for every single device detail page visit. Let's return a BAD_REQUEST 401 and at the UI side will tackle this error?